Content.Shared/Identity/Systems/IdentitySystem.cs

```diff
@@ -0,0 +1,220 @@
+﻿using Content.Shared.Access.Systems;
+using Content.Shared.CharacterAppearance.Components;
+using Content.Shared.Ghost;
+using Content.Shared.Inventory;
+using Content.Shared.Preferences;
+using Robust.Shared.Enums;
+
+namespace Content.Shared.Identity.Systems;
+
+/// <summary>
+///     Handles getting an entity's 'identity', essentially an IC version of <see cref="MetaDataComponent.EntityName"/>
+/// </summary>
+public sealed partial class IdentitySystem : EntitySystem
+{
+    [Dependency] private readonly SharedIdCardSystem _sharedIdCardSystem = default!;
+
+    /// <summary>
+    ///     Gets an 'identity string' representing an entity's name from the viewer's perspective.
+    /// </summary>
+    /// <param name="target">The entity to get an identity string for.</param>
+    /// <param name="viewer">The "viewer" of the identity.</param>
+    /// <param name="useTrueName">Should we get the entity's true name, if it's known?</param>
+    /// <param name="inventory">Resolve comp</param>
+    /// <param name="appearance">Resolve comp</param>
+    public string GetIdentityString(EntityUid target, EntityUid? viewer=null, bool useTrueName=false,
+        InventoryComponent? inventory=null,
+        HumanoidAppearanceComponent? appearance=null)
+    {
+        if (!Resolve(target, ref inventory, false))
+            return Name(target);
+
+        // These events handles things like masks blocking identity, or monkeys
+        // not knowing identity.
+
+        var viewerEv = new ShouldKnowIdentityAttemptEvent(target);
+        // What the viewer says
+        if (viewer != null)
+        {
+            RaiseLocalEvent(viewer.Value, viewerEv, false);
+        }
+
+        var targetEv = new CanKnowIdentityAttemptEvent(viewer);
+        RaiseLocalEvent(target, targetEv, false);
+
+        var trueName = useTrueName || viewerEv.KnowsTrueName || targetEv.KnowsTrueName;
```

Comment:
If you find out someone's True Name, does it give you insight into their very nature, and thus grant absolute control?
https://en.wikipedia.org/wiki/True_name
<./memeing>